lib/spack/spack/test/util/executable.py

```diff
@@ -62,24 +75,49 @@ def test_which_with_slash_ignores_path(tmpdir, working_env):
     wrong_path = str(tmpdir.join('bin', 'exe'))
     os.environ['PATH'] = os.path.dirname(wrong_path)
 
-    fs.set_executable(path)
-    fs.set_executable(wrong_path)
+    if sys.platform == "win32":
+        with tmpdir.as_cwd():
+            fs.touch("exe.exe")
```

Comment:
Instead of using `fs.touch()` here, add an if `sys.platform == "win32": `above and use `tmpdir.ensure('exe.exe') `instead

Reply:
Same question as above.